packages/ember-metal/lib/computed.js

```diff
@@ -426,11 +426,11 @@ ComputedPropertyPrototype._set = function computedPropertySet(obj, keyName, valu
 };
 
 /* called before property is overridden */
-ComputedPropertyPrototype.teardown = function(obj, keyName) {
+ComputedPropertyPrototype.teardown = function(obj, keyName, _meta) {
   if (this._volatile) {
     return;
   }
-  let meta = metaFor(obj);
```

Comment:
yup this is correct, but we will now need to guard against `meta === null`.

something like
```js
+  let meta = _meta || peekMeta(obj);
 
if ( meta === null) { return; }
```